### PR TITLE
DEV-101 badges views

### PR DESCRIPTION
**Pull request by matgrube**

### CODE REVIEW

File: app/src/main/java/pl/edu/pjatk/goodwill_ninjas/blooddonor_android/api/userFeat/UserFeatResponse.kt [hunk 1]
```diff
@@ -6,7 +6,16 @@ data class UserFeatResponse(
     @JsonProperty("userId") val userId: Int = 0,
     @JsonProperty("featId") val featId: String = "",
     @JsonProperty("featName") val featName: String = "",
-    @JsonProperty("featDescription") val featDescription: String = "",
-    @JsonProperty("achievedRanks") val achievedRanks: String = "",
-    @JsonProperty("nextRanks") val nextRanks: String = ""
+    @JsonProperty("featDescription") val featDescription: String? = "",
+    @JsonProperty("achievedRanks") val achievedRanks: List<UserFeatRanks> = listOf(UserFeatRanks()),
+    @JsonProperty("nextRanks") val nextRanks: List<UserFeatRanks> = listOf(UserFeatRanks())
```
Comment (s-kakol) [inline app/src/main/java/pl/edu/pjatk/goodwill_ninjas/blooddonor_android/api/userFeat/UserFeatResponse.kt:11]: These should be in snake case coming from the API. Want me to fix that or are we sticking with this for the time being?

File: app/src/main/java/pl/edu/pjatk/goodwill_ninjas/blooddonor_android/components/mainPageBadge/MainPageBadge.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            image,
+            "odznaka",
+            modifier = Modifier
+                .aspectRatio(ratio =
+                image.intrinsicSize.height /
+                        image.intrinsicSize.width)
+                .fillMaxWidth()
+                .fillMaxHeight(),
+            contentScale = ContentScale.Fit,
+            colorFilter = ColorFilter.colorMatrix(grayScaleMatrix)
```
Comment (s-kakol) [inline app/src/main/java/pl/edu/pjatk/goodwill_ninjas/blooddonor_android/components/mainPageBadge/MainPageBadge.kt:45]: If for 0 and 1 could be merged and this grayed out line singled out but it's me being nitpicky

**Review by s-kakol — APPROVED** [review]
LGTM, approved but please look at the comment regarding the API response

File: app/src/main/java/pl/edu/pjatk/goodwill_ninjas/blooddonor_android/components/mainPageBadge/MainPageBadge.kt (hunk 2 above)
Comment (matgrube) [reply app/src/main/java/pl/edu/pjatk/goodwill_ninjas/blooddonor_android/components/mainPageBadge/MainPageBadge.kt:45]: Will fix in next task, I'm using that function there as well

**Pull request merged**